zen-quotes-app/src/app/home/helpers.ts

```diff
@@ -1,8 +1,32 @@
-// @ts-ignore
 // @Todo
 //  @Ira: when we run `ng test`, the tests in helpers.spect.ts fail.
 //  Can you implement this function so that the tests succeed?
 //  Thanks and best of luck! 🍀
-export function makeAuthorsUnique(authors) {
-  return authors;
+import { orderBy, uniqBy } from 'lodash';
+
+import { Author } from '../../types/author.type';
+import { Quote } from '../../types/quote.type';
+
+export function mapAuthorsFromQuotes(quotes: Quote[] | []): Author[] {
+  return quotes
+    ? quotes.map((quote) => {
+        return { name: quote.author };
+      })
+    : [];
+}
+
+export function makeAuthorsUnique(authors: Author[] | undefined): Author[] | [] {
+  return authors ? uniqBy(authors, 'name') : [];
+}
+
+export function sortAuthors(authors: Author[] | undefined): Author[] | [] {
+  return authors ? orderBy(authors, 'name', ['asc']) : [];
+}
+
+export function getAuthors(quotes: Quote[] | undefined): Author[] | [] {
+  return quotes ? sortAuthors(makeAuthorsUnique(mapAuthorsFromQuotes(quotes))) : [];
+}
+
+export function generateRandomQuote(quotes: Quote[] | undefined): Quote {
+  return quotes ? quotes[Math.floor(Math.random() * quotes.length)] : { author: '', text: '' };
 }
```

Comment:
Hi @jvandemo 
Could you take a look when you have time, please 🙏🏻
Does it look better and more testable now? Sorry if it looks incorrect again.
Honestly, I am unsure if I can do this in the getAuthors function, but I don't know how to do it another way for now. 🤷

Thank you in advance!🙏🏻🙇‍♀️

Reply:
@IraLysenko This looks great! 🤩  Awesome work! 👏 

You created functions that take data as input and return data as output. That is wonderful. That makes the function decoupled (not dependent on state outside of the function) and makes them much easier to test indeed.

⚡ Next challenge:
can you add unit tests for `makeAuthorsUnique`, `sortAuthors`, `getAuthors`, and `getAuthors`?

Keep rocking—your code is so much better already! 🏆

Reply:
@jvandemo 
Hooray!Thank you very much!🙏🏻🙇‍♀️
Good, I will add it!

Thank you very much for your time and guidance!🙏🏻🙇‍♀️

Reply:
🏆 ✨